docs/userguide/constraints.md

```diff
@@ -222,10 +233,123 @@ DiscreteLinkedParametersConstraint(
+Let's say an experimental setup has a parameter called "Switch" which turns on 
+pieces of equipment that are optional.
+This means the other parameters (called ``affected_parameters``) are only relevant if 
+the "Switch" parameter has the value "on". If the switch is "off", the affected parameters 
+are irrelevant.
+
+You can specify such a dependency with the ``DiscreteDependenciesConstraint``, which 
+requires:
+1) A list ``parameters`` with the names of the parameters upon which others depend.
+2) A list of ``Condition``'s which specify the values of the corresponding entries in 
+   ``parameters`` which "activate" the dependent parameters.
+3) A list of lists, each containing the ``affected_parameters``, which become relevant 
+   only if the corresponding entry in ``parameters`` is active as specified by the entry in ``conditions``.
+
+Internally, BayBE drops entries from the ``SearchSpace`` where affected parameters are 
+irrelevant. Since in our example "off" is still a valid value for the switch, the 
+``SearchSpace`` will still retain an entry for that combination, showing arbitrary 
+values for the ``affected_parameters`` (which can be ignored).
+
+(DDC)=
+```{note}
+BayBE requires that all dependencies are declared in a single 
+``DiscreteDependenciesConstraint``, creating a ``SearchSpace`` from multiple 
+``DiscreteDependenciesConstraint``'s will throw a validation error.
+```
+
+In the example below we mimic a situation where there are two switches and each switch 
+activates two other parameters that are only relevant if the first switch is "on" / the 
+second switch is on "right" respectively.
+```python
+from baybe.constraints import DiscreteDependenciesConstraint, SubSelectionCondition
+
+DiscreteDependenciesConstraint(
+    parameters=["Switch1", "Switch2"],  # the two parameters upon which others depend
+    conditions=[
+        SubSelectionCondition(selection=["on"]),     # values of Switch1 that activate the affected parameters
+        SubSelectionCondition(selection=["right"]),  # values of Switch2 that activate the affected parameters
+    ],
+    affected_parameters=[
+      ["Solvent", "Fraction"],  # parameters affected by "Switch1"
+      ["Frame1", "Frame2"]      # parameters affected by "Switch2"
+    ],
+)
+```
+
+An end to end example can be found [here](../../examples/Constraints_Discrete/dependency_constraints).
 
 ### ``DiscretePermutationInvarianceConstraint``
-Content coming soon...
+Permutation invariance is a property where combinations of values of multiple
+parameters do not depend on their order due to some symmetry in the experiment. 
+Suppose that we are creating a mixture which has up to three solvents, i.e. parameters 
+"Solvent 1", "Solvent 2", "Solvent 3".
+In this situation, all combinations from the following table would be equivalent, 
+hence the ``SearchSpace`` should effectively only contain one of them.
+
+|   | Solvent 1    | Solvent 2    | Solvent 3    |
```

Comment:
these three sovlents all have their separate encodings. This becomes especially relevant if the mixture components are not the same type or have different possible values, so in general they might have different encodings. 

Putting them all in one parameter takes away all the advantages of encoding them smartly. I've never heard of someone putting all of these into one, maybe its a viable alternative, which however does not really mean this one used here is invalid.

Reply:
The reason is that since the upper table only contains permutations of the same three substances, I did not think of situations in which there are e.g. only two substances mixed together. So this is actually not an issue and I simply had that question due to mis-reading/mis-interpreting.